evalml/preprocessing/utils.py

```diff
@@ -144,6 +149,23 @@ def split_data(
         1    9
         dtype: int64
     """
+    if is_multiseries(problem_type) and isinstance(y, pd.Series):
+        series_id = problem_configuration.get("series_id")
```

Comment:
Should check if problem_configuration exists and key exists before getting and if not also raise the same value error.